torch/_inductor/pattern_matcher.py

```diff
@@ -342,17 +342,29 @@ def __repr__(self):
         return f"{self.__class__.__name__}({', '.join(args)})"
 
     def _match(self, node: torch.fx.Node, ctx: MatchContext):
-        if (
-            not self._match_fns(node)
-            or len(node.args) != len(self.args)
-            or len(node.kwargs) != len(self.kwargs)
-        ):
+        if not self._match_fns(node):
             return FailedMatch(f"function_mismatch: node={node}, pattern={self}")
 
         if not self._match_users(node, ctx):
             return FailedMatch(f"multiple_users {node}")
 
-        node_items, node_spec = self.flatten(node.args, node.kwargs)
+        _args = node.args
+        _kwargs = node.kwargs
+        if len(_args) != len(self.args) or len(_kwargs) != len(self.kwargs):
```

Comment:
Should we make sure that `self.args` and `self.kwargs` are also normalized?

Reply:
I was thinking about this, but ```self.args``` and ```self.kwargs``` are from the the pattern user defined, where we can ensure they are normalized. And I'm not sure if we allow users intend to write unnormalized pattern in some scenarios.

Reply:
[Here](https://github.com/pytorch/pytorch/blob/main/torch/_inductor/fx_passes/freezing_patterns.py#L153-L155), for example, `alpha` and `beta` are not explicitly specified. I guess, the pattern author meant to use the defaults, but then we have an unnormalized pattern?